pkg/operator/ceph/object/user.go

```diff
@@ -266,6 +269,10 @@ func generateCephUserSecret(userConfig *admin.User, endpoint, namespace, storeNa
 		StringData: secrets,
 		Type:       k8sutil.RookType,
 	}
+	if UserKeysSource {
```

Comment:
Is this using a global variable in a reconcile? If so, that is a big no-no. We really should never do this, with almost no exceptions.

Reply:
Should I return the value throught the functions and pass this using function parameters?

Reply:
I thought it should be a CRD setting on the objectstoreuser

Reply:
Maybe right, but then user has to update at two different places.

Reply:
What's the second place? It should only be a CRD setting unless I'm missing something.

Reply:
So user has to do 2 steps,

1) Update the CRD with the annotations.
2) Update the k8s secret manually with new keys.

The current approach, we were adding annotation and updating the keys of the object user secret

Reply:
Oh right, we don't need a CRD setting, we are just using the annotation on the secret. So the flow is:
1. Create the k8s secret with desired keys, and the annotation that the secret is the source of truth.
2. Create the CRD for the CephObjectStoreUser, as usual (no change)

But I think there is still confusion why we need a global var `UserKeysSource`. The annotation on the secret would tell us that info.

Reply:
>But I think there is still confusion why we need a global var UserKeysSource. The annotation on the secret would tell us that info.

The function `func (r *ReconcileObjectStoreUser) createOrUpdateCephUser(u *cephv1.CephObjectStoreUser) error {
` unfold whether the secret has the correct annotations or not.

And the function `func generateCephUserSecret(userConfig *admin.User, endpoint, namespace, storeName, tlsSecretName string) *corev1.Secret { `
which is later called in the reconcile actually formats the secret and `createandupdate` it,

So, 
1) I choose to have global variable between these two functions,
Alternative I can think of
2)  `createOrUpdateCephUser` function should return the value whether secret has the annotation `secretannotationexiste`
3) function `generateCephUserSecret` refetch the secret.

Maybe you can help me understand the best practice

Reply:
Per separate discussion, we can use a local return value to avoid the package var.